controllers/dockermachine_controller_test.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+	"k8s.io/klog/klogr"
+	infrav1 "sigs.k8s.io/cluster-api-provider-docker/api/v1alpha2"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+)
+
+func init() {
+	klog.InitFlags(nil)
+}
+
+func setupScheme() *runtime.Scheme {
+	s := runtime.NewScheme()
+	if err := clusterv1.AddToScheme(s); err != nil {
+		panic(err)
+	}
+	if err := infrav1.AddToScheme(s); err != nil {
+		panic(err)
+	}
+	return s
+}
+
+func TestDockerMachineReconciler_DockerClusterToDockerMachines(t *testing.T) {
+	clusterName := "my-cluster"
+	dockerCluster := newDockerCluster(clusterName, "my-docker-cluster")
+	dockerMachine1 := newDockerMachine("my-docker-machine-0")
+	dockerMachine2 := newDockerMachine("my-docker-machine-1")
+	objects := []runtime.Object{
+		newCluster(clusterName),
+		dockerCluster,
+		newMachine(clusterName, "my-machine-0", dockerMachine1),
+		newMachine(clusterName, "my-machine-1", dockerMachine2),
+		// Intentionally omitted
+		newMachine(clusterName, "my-machine-2", nil),
+	}
+	c := fake.NewFakeClientWithScheme(setupScheme(), objects...)
+	r := DockerMachineReconciler{
+		Client: c,
+		Log:    klogr.New(),
+	}
+	mo := handler.MapObject{
+		Object: dockerCluster,
+	}
+	out := r.DockerClusterToDockerMachines(mo)
+	if len(out) != 2 {
```

Comment:
Is there value in checking that you got the correct keys by name?

Reply:
I'd be surprised if they were duplicated but this is an easy enough check! I'd rather have a test for it here then some weird bug later when I'm working on something else

Reply:
👍 I tend to get leery when unit tests check expected lengths where the number is 2 or greater when the specific entries in the list matter

Reply:
fixed!